Add MySQL scaler docs (#72)
kedacore/keda-docs: content/scalers/mysql.md

+++
fragment = "content"
weight = 100
title = "MySQL"
background = "light"
+++

Scale applications based on MySQL query result.

> **Availability:** v1.1+ | **Maintainer:** Community

Reviewer: This is actually not correct since 1.1 has already been shipped.

```suggestion
**Availability:** v1.2+ | **Maintainer:** Community
```

<!--more-->

### Trigger Specification

This specification describes the `mysql` trigger that scales based on result of MySQL query.

> The trigger requires the following information:

Reviewer: Let's add a newline to improve readability
```suggestion

The trigger requires the following information:
```

> - `query` - a MySQL query that should return single numeric value that will be used for HPA

Reviewer: We should add a line as the list is not rendered correctly, see https://deploy-preview-72--vigorous-franklin-8edd65.netlify.com/scalers/mysql/
```suggestion

- `query` - a MySQL query that should return single numeric value that will be used for HPA
```

> - `query` - a MySQL query that should return single numeric value that will be used for HPA

Reviewer: HPA is implementation detail, not everybody will be aware what it is since we abstract that out.
```suggestion
- `query` - a MySQL query that should return single numeric value that will be used for scaling
```

Alternatively, we can even shorten it more
```suggestion
- `query` - a MySQL query that should return single numeric value
```

- `queryValue` - a threshold that is used as `targetAverageValue` in HPA. 

> - Connection information as `dsnConnectionStr` or set of required params (more information below)

Reviewer: Nitpick - Wouldn't it be better to call it `connectionString`?

Author: Yes, it was suggested also in PR with the scaler :)

#### Using `dsnConnectionStr`
One option to connect to MySQL database is to use dsn connection string from environment variable:
```yaml
triggers:
  - type: mysql
    metadata:

>       dsnConnectionStr: "SQL_CONN_STR" // SQL_CONN_STR env variable should point to valid connection string

Reviewer: I would make it so that `dsnConnectionStr` can be configured via `TriggerAuthentication` as well

Author: It will work also with `TriggerAuthentication`

      queryValue: "4"
      query: "SELECT CEIL(COUNT(*) / 6) FROM task_instance WHERE state='running' OR state='queued'"
```

#### Using authentication parameters
Other possibility is to pass all required parameters to build a new connection string:
```yaml
triggers:
  - type: mysql
    metadata:
      username: "root"
      password: "MYSQL_PASSWORD" // MYSQL_PASSWORD env variable should point to the password
      host: "mysql"
      port: "3306"
      dbname: "stats_db"
      queryValue: "4"
      query: "SELECT CEIL(COUNT(*) / 6) FROM task_instance WHERE state='running' OR state='queued'"
```
In this case the following additional parameters must be passed:

> - `username` used to access MySQL database

Reviewer: All of these should be documented here: https://github.com/kedacore/keda-docs/pull/72/files#diff-0015a92fc8bd54f11e712f524fd72fc9R20

Reviewer: We tend to list all of the optional fields in one list so that folks know what to use when

- `password` used for the given user, this should be blank (no password) or point to an environment
 variable with the password
- `host` and `port` of the database
- `dbname` as name of the database

### Example

> Here is an example of how to deploy a scaled object with the `mysql` scale trigger which uses

Reviewer: Personally I'd go as secure/simple as possible and use `connectionString` approach here, thoughts?

Author: I am ok with that ;)

+++
fragment = "content"
weight = 100
title = "MySQL"
background = "light"
+++

Scale applications based on MySQL query result.

**Availability:** v1.2+ | **Maintainer:** Community

<!--more-->

### Trigger Specification

This specification describes the `mysql` trigger that scales based on result of MySQL query.

The trigger always requires the following information:

- `query` - a MySQL query that should return single numeric value
- `queryValue` - a threshold that is used as `targetAverageValue` in HPA.

> to provide information about how to connect to MySQL you can provide 

Reviewer: ```suggestion
To provide information about how to connect to MySQL you can provide 
```

- `connectionString` MySQL connection string that should point to environment variable with valid value

> or

Reviewer: ```suggestion
Or provide all the required information as part of the trigger metadata:
```

- `username` used to access MySQL database
- `password` used for the given user, this should be blank (no password) or point to an environment
 variable with the password
- `host` and `port` of the database
- `dbName` as name of the database

#### Using `connectionString`
One option to connect to MySQL database is to use dsn connection string from environment variable:
```yaml
triggers:
  - type: mysql
    metadata:

>       connectionString: "SQL_CONN_STR" // SQL_CONN_STR env variable should point to valid connection string

Reviewer: YAML uses `#` for comments
```suggestion
      connectionString: "SQL_CONN_STR" # SQL_CONN_STR env variable should point to valid connection string
```

      queryValue: "4"
      query: "SELECT CEIL(COUNT(*) / 6) FROM task_instance WHERE state='running' OR state='queued'"
```

#### Using authentication parameters
Other possibility is to pass all required parameters to build a new connection string:
```yaml
triggers:
  - type: mysql
    metadata:
      username: "root"

>       password: "MYSQL_PASSWORD" // MYSQL_PASSWORD env variable should point to the password

Reviewer: YAML uses `#` for comments
```suggestion
      password: "MYSQL_PASSWORD" # MYSQL_PASSWORD env variable should point to the password
```

+++
fragment = "content"
weight = 100
title = "MySQL"
background = "light"
+++

Scale applications based on MySQL query result.

**Availability:** v1.2+ | **Maintainer:** Community

<!--more-->

### Trigger Specification

This specification describes the `mysql` trigger that scales based on result of MySQL query.

The trigger always requires the following information:

- `query` - a MySQL query that should return single numeric value
- `queryValue` - a threshold that is used as `targetAverageValue` in HPA.

To provide information about how to connect to MySQL you can provide 
- `connectionString` MySQL connection string that should point to environment variable with valid value

> Or

Reviewer: ```suggestion
Or provide more detailed information:
```

- `username` used to access MySQL database
- `password` used for the given user, this should be blank (no password) or point to an environment
 variable with the password
- `host` and `port` of the database
- `dbName` as name of the database

### Authentication Parameters

> Both `connectionString` and `password` can be configured by `TriggerAuthentication` CRD or as environment

Reviewer: Let's align this with other scalers:
```suggestion
You can authenticate by using connection string or password authentication.

**Connection String Authentication:**

- `connection` - Connection string for MySQL database

**Password Authentication:**

- `connection` - Password for configured user to login to MySQL database
```
